--- linux-4.14.18/arch/arm/mach-pxa/spitz.c	2018-02-11 14:35:13.268748129 +1300
+++ linux-4.14.18-fb/arch/arm/mach-pxa/spitz.c	2018-06-01 17:21:11.952522957 +1200
@@ -15,6 +15,7 @@
 #include <linux/kernel.h>
 #include <linux/module.h>	/* symbol_get ; symbol_put */
 #include <linux/platform_device.h>
+#include <linux/memblock.h>
 #include <linux/delay.h>
 #include <linux/gpio_keys.h>
 #include <linux/gpio.h>
@@ -34,6 +35,7 @@
 #include <linux/reboot.h>
 #include <linux/memblock.h>
 
+#include <asm/sizes.h>
 #include <asm/setup.h>
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
@@ -1024,9 +1026,14 @@
 	memblock_add(0xa0000000, SZ_64M);
 }
 
+static void __init Cxx0_reserve_fbmem (void) {
+	memblock_reserve(0xA3E00000, SZ_1M);
+}
+
 #ifdef CONFIG_MACH_SPITZ
 MACHINE_START(SPITZ, "SHARP Spitz")
 	.fixup		= spitz_fixup,
+	.reserve	= Cxx0_reserve_fbmem,
 	.map_io		= pxa27x_map_io,
 	.nr_irqs	= PXA_NR_IRQS,
 	.init_irq	= pxa27x_init_irq,
@@ -1040,6 +1047,7 @@
 #ifdef CONFIG_MACH_BORZOI
 MACHINE_START(BORZOI, "SHARP Borzoi")
 	.fixup		= spitz_fixup,
+	.reserve	= Cxx0_reserve_fbmem,
 	.map_io		= pxa27x_map_io,
 	.nr_irqs	= PXA_NR_IRQS,
 	.init_irq	= pxa27x_init_irq,
@@ -1053,6 +1061,7 @@
 #ifdef CONFIG_MACH_AKITA
 MACHINE_START(AKITA, "SHARP Akita")
 	.fixup		= spitz_fixup,
+	.reserve	= Cxx0_reserve_fbmem,
 	.map_io		= pxa27x_map_io,
 	.nr_irqs	= PXA_NR_IRQS,
 	.init_irq	= pxa27x_init_irq,
